### PR TITLE
New section in the indexing docs about identifying single rows.

### PR DESCRIPTION
**Pull request by tom-clickhouse**

### CODE REVIEW

File: docs/en/guides/improving-query-performance/sparse-primary-indexes/sparse-primary-indexes-uuids.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+# Identifying single rows efficiently
+
+Although in general it is [not](https://clickhouse.com/docs/en/faq/use-cases/key-value) the best use case for ClickHouse, 
+sometimes applications built on top of ClickHouse require to identify single rows of a ClickHouse table. 
+
+ 
+An intuitive solution for that might be to use a [UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier) column with a unique value per row and for fast retrieval of rows to use that column as a primary key column.
+
+For the fastest retrieval, the UUID column [would need to be first key column](./sparse-primary-indexes-design#the-primary-index-is-used-for-selecting-granules).
```
Comment (DanRoscigno) [inline docs/en/guides/improving-query-performance/sparse-primary-indexes/sparse-primary-indexes-uuids.md:15]:
```suggestion
For the fastest retrieval, the UUID column [would need to be the first key column](./sparse-primary-indexes-design#the-primary-index-is-used-for-selecting-granules).
```

File: docs/en/guides/improving-query-performance/sparse-primary-indexes/sparse-primary-indexes-uuids.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+---
+sidebar_label: Identifying single rows efficiently
+sidebar_position: 5
+description: TODO
```
Comment (rfraposa) [inline docs/en/guides/improving-query-performance/sparse-primary-indexes/sparse-primary-indexes-uuids.md:4]: Needs a description...

File: docs/en/guides/improving-query-performance/sparse-primary-indexes/sparse-primary-indexes-uuids.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+And one way to identify and retrieve (a specific version of) the pasted content is to use a hash of the content as the UUID for the table row that contains the content.
+
+The following diagram shows 
+- the insert order of rows when the content changes (for example because of keystrokes typing the text into the text-area) and
+- the on-disk order of the data from the inserted rows when the `PRIMARY KEY (hash)` is used:
+<img src={require('./images/sparse-primary-indexes-15a.png').default} class="image"/>
+
+Because the `hash` column is used as the primary key column
+- specific rows can be retrieved [very fast](./sparse-primary-indexes-design#the-primary-index-is-used-for-selecting-granules), but
+- the table's rows (their column data) is stored on disk ordered ascending by (the unique and random) hash values. Therefore also the content column's values are stored in random order with no data locality resulting in a **suboptimal compression ratio for the content column data file**.
```
Comment (rfraposa) [inline docs/en/guides/improving-query-performance/sparse-primary-indexes/sparse-primary-indexes-uuids.md:36]: "is stored" should be "are stored"

**Review by DanRoscigno — APPROVED** [review]
A couple of small changes, thanks for this!

**Pull request merged**